src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1056,10 +1056,17 @@ impl Cage {
     *   - null, fail 
     */
     pub fn getcwd_syscall(&self, buf: *mut u8, bufsize: u32) -> i32 {
+        if (!buf.is_null() && bufsize == 0) || (buf.is_null() && bufsize != 0) {
+            unsafe { *libc::__errno_location() = libc::EINVAL };
```

Comment:
`*libc::__errno_location()` will catch the error with their number and location from libc. We shouldn't assign the value to it inside RawPOSIX. Instead, we need to set error returns appropriately by ourselves by using `syscall_error` function in `/interface/errno.rs`.

Reply:
Updated to use: `syscall_error`